### PR TITLE
Add Encoding::Converter signature and tests

### PR DESCRIPTION
**Pull request by snaka**

### CODE REVIEW

File: test/stdlib/Encoding_Converter_test.rb [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+      Encoding::Converter.new("EUC-JP", "ISO-8859-1"), :primitive_convert,
+      "\xa4", "", nil, 10, Encoding::Converter::PARTIAL_INPUT
+    )
+    assert_send_type(
+      "(String, String, nil, Integer, partial_input: true) -> :source_buffer_empty",
+      Encoding::Converter.new("EUC-JP", "ISO-8859-1"), :primitive_convert,
+      "\xa4", "", nil, 10, partial_input: true
+    )
+
+    # TODO: Add after_output test case
```
Comment (snaka) [inline test/stdlib/Encoding_Converter_test.rb:296]: I don't know what cases return `:after_output`, so I haven't implemented that test.

File: test/stdlib/Encoding_Converter_test.rb [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+    assert_send_type(
+      "(self) -> bool",
+      Encoding::Converter.new("UTF-8", "EUC-JP"), :==, Encoding::Converter.new("UTF-8", "EUC-JP")
+    )
+  end
+
+  # def test_initialize
+  #   assert_send_type  "(*untyped) -> void",
+  #                     Encoding::Converter.new, :initialize
+  # end
```
Comment (snaka) [inline test/stdlib/Encoding_Converter_test.rb:194]: I think `ConverterSingletonTest#test_new` covers the `#initializer` test, so no test is needed here.

**Comment by snaka** [comment]
Linked
https://github.com/ruby/rbs/issues/1

**Review by soutaro — APPROVED** [review]
Thanks!

**Pull request merged**